### DISCUSSION TITLE
easy way to have a page show a list of links for all child pages?

### DISCUSSION THREAD

**Discussion by mdougher** [redimp/otterwiki#112] [General]
so i organize my pages like this:
foo
foo/bar
foo/bar/baz
where "baz" has some real content, but "foo" and "bar" are more just like containers with no content of their own.
this leaves those pages basically just empty. I would like to have those pages display links for all of the child pages in their hierarchy.
like this:
foo has a link for bar and bar/baz.
bar has a link for baz.

Basically, the same kind of structure shown in the left sidebar, but only showing the structure for this page and anything below it.

Yes, i can add the links myself, but that could be time consuming.
Is there any simpler path to creating those links?
Also, if new child pages are added, i would have to come back and update the links, which is more maintenance.
something to automatically figure out what the child pages are when the page is displayed would be awesome.

is something like this possible, or am i dreaming a little too far?

**Comment by redimp** [comment]
Hey @mdougher, currently the only way to achive an overview over the child pages (except for the sidebar) is an empty folder with child pages, e.g.

- Without adding a `foo` or `foo/bar` create a page `foo/bar/baz`.
- `/foo` will show a page index with `bar` and `baz` below it:
  <img width="193" alt="image" src="https://github.com/redimp/otterwiki/assets/112056/4597993e-74bd-47d9-814f-97c5c43402c0">
- `/foo/bar` will show a page index with `baz` below it:
  <img width="191" alt="image" src="https://github.com/redimp/otterwiki/assets/112056/f87a8b10-13e5-4fc8-88b7-fc416da60425">

But with content in e.g. `foo/bar` the page index will not be shown. There is currently no feature to add this. 

Before the child pages where shown in the sidebar, which seems to be sufficient for the most purposes, I thought about two options to approach this:

1.) Adding the feature of more special blocks to the syntax, e.g.

```
:::pageindex
:::
```

2.) Adding the feature of a yaml front matter with the option to add the pageindex (and e.g. tags):

```
---
pageindex: top
tags: example, baz
---
# Baz

Example
```

**Reply by mdougher** [reply]
thats perfect! thanks so much!
